components/styled/table/src/molecules/useScrollRef.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react'
+
+const useScrollRef = (elementRef: React.RefObject<HTMLElement>) => {
+  const [scrollState, setScrollState] = useState<ScrollState>()
+  const element = elementRef.current as HTMLElement
```

Comment:
We shouldn't be casting variables into other types to get around null checks - this will cause an error if this hook ever genuinely receives a null argument